tldw_Server_API/app/core/TTS/tts_validation.py

```
@@ -624,6 +634,13 @@ def validate_request(self, request: TTSRequest, provider: Optional[str] = None)
                         )
                     if min_duration is None:
                         min_duration = 3.0
+            elif provider == "pocket_tts_cpp":
+                voice = (request.voice or "").strip()
+                if not request.voice_reference and not voice.startswith("custom:"):
+                    raise TTSInvalidVoiceReferenceError(
+                        "PocketTTS.cpp requires a direct voice_reference or a stored custom: voice",
+                        provider=provider,
+                    )
```

Ruff (0.15.7)

[warning] Abstract `raise` to an inner function

(TRY301)